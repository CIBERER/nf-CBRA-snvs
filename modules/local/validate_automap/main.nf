process VALIDATE_AUTOMAP {
    tag "$meta.id"
    input:
    tuple val(meta), path(vcf)
   
    output:
    tuple val(meta), path(vcf), env('VALIDATION_STATUS'), emit: validate_automap
   
    script:
    """
    line_count=\$(zgrep -v "#" ${vcf} | wc -l)
    if [ \$line_count -ge 1000 ]; then
        VALIDATION_STATUS="PASS"
        echo "File validation passed"
    else
        VALIDATION_STATUS="FAIL"
        echo "File validation failed: only \$line_count lines"
    fi
    """
}
