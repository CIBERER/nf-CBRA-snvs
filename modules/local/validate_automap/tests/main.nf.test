// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test validate_automap
nextflow_process {

    name "Test Process VALIDATE_AUTOMAP"
    script "../main.nf"
    process "VALIDATE_AUTOMAP"

    tag "modules"
    tag "modules_nfcore"
    tag "validate_automap"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("validate_automap - vcf") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("https://raw.githubusercontent.com/mquinodo/AutoMap/refs/heads/master/Test/TestSample.vcf", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

}
