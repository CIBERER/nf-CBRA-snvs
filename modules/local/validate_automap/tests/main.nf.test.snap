{
    "validate_automap - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "TestSample.vcf:md5,c9fd342f1691361a843356149e24f901",
                        "PASS"
                    ]
                ],
                "validate_automap": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "TestSample.vcf:md5,c9fd342f1691361a843356149e24f901",
                        "PASS"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-23T10:31:45.384814062"
    }
}